services/api/src/api/routes/endpoint.py

```diff
@@ -28,21 +30,124 @@
+
 
 class EndpointsDefinition:
     """Definition of supported endpoints and its relation with processing steps."""
 
-    processing_steps_by_endpoint: Mapping[str, List[ProcessingStep]]
+    processing_steps_by_endpoint: StepsByInputTypeAndEndpoint
 
     def __init__(self, graph: ProcessingGraph, endpoint_config: EndpointConfig):
-        self.definition = {
-            endpoint: [graph.get_step(step) for step in processing_steps]
-            for endpoint, processing_steps in endpoint_config.specification.items()
+        self.processing_steps_by_endpoint = {
+            endpoint: {
+                input_type: [graph.get_step(step) for step in steps] for input_type, steps in input_types.items()
+            }
+            for endpoint, input_types in endpoint_config.specification.items()
         }
 
 
-def create_endpoint(
+@dataclass
+class InputParameters:
+    dataset: str
+    config: Optional[str]
+    split: Optional[str]
+    input_type: InputType = field(init=False)
+
+    def __post_init__(self) -> None:
+        self.input_type = "split" if self.split else "config" if self.config else "dataset"
+
+
+def get_input_parameters(query_params: QueryParams) -> InputParameters:
+    dataset = query_params.get("dataset")
+
+    if not are_valid_parameters([dataset]) or not dataset:
+        raise MissingRequiredParameterError("Parameter 'dataset' is required")
+
+    config = None
+    config_param = query_params.get("config")
+    if are_valid_parameters([config_param]):
+        config = config_param
+
+    split = None
+    split_param = query_params.get("split")
+    if are_valid_parameters([split_param]):
+        split = split_param
+        if config is None:
+            raise MissingRequiredParameterError("Parameter 'config' is required")
+
+    return InputParameters(dataset=dataset, config=config, split=split)
+
+
+def get_first_succeeded_cache_entry_from_steps(
     processing_steps: List[ProcessingStep],
+    input_parameters: InputParameters,
+    init_processing_steps: List[ProcessingStep],
+    hf_endpoint: str,
+    hf_token: Optional[str] = None,
+) -> Optional[CacheEntry]:
+    """Gets the cache from the first succedeed step in the processing step list.
+    If no succeeded result is found, it will return the last one even if it was failed,
+    if no one result is found, it returns None.
+    Checks if job is still in progress by each processing step in case of no entry found.
+    """
+    last_result = None
+    for processing_step in processing_steps:
+        try:
+            last_result = get_response(
+                kind=processing_step.cache_kind,
+                dataset=input_parameters.dataset,
+                config=input_parameters.config,
+                split=input_parameters.split,
+            )
+
+            if last_result["http_status"] == HTTPStatus.OK:
+                return last_result
+        except DoesNotExist:
+            logging.warning(
```

Comment:
maybe not a warning, because it will be common to have missing cache entries (the job has not finished yet). Info or debug

Reply:
I changed it to debug